stdpopsim/slim_engine.py

```diff
@@ -849,12 +849,28 @@ def matrix2str(
     # Mutation types.
     for i, m in enumerate(mutation_types):
         distrib_args = [str(arg) for arg in m.distribution_args]
-        distrib_args[m.Q_scaled_index] = "Q * " + distrib_args[m.Q_scaled_index]
-        distrib_args = ", ".join(distrib_args)
-        printsc(
-            f"    initializeMutationType({i}, {m.dominance_coeff}, "
-            + f'"{m.distribution_type}", {distrib_args});'
-        )
+        if m.Q_scaled_index > -1:
+            distrib_args[m.Q_scaled_index] = "Q * " + distrib_args[m.Q_scaled_index]
+            distrib_args = ", ".join(distrib_args)
+            printsc(
+                f"    initializeMutationType({i}, {m.dominance_coeff}, "
+                + f'"{m.distribution_type}", {distrib_args});'
+            )
+        # dealing with scaling a lognormal distribution
+        # (adding instead of multiplying the sd):
+        else:
+            if scaling_factor == 1:
+                distrib_args = (
+                    f'"return rlnorm(1, {distrib_args[0]}, {distrib_args[1]});"'
+                )
+            else:
+                distrib_args = (
+                    f'"return rlnorm(1, {distrib_args[0]}, Q + {distrib_args[1]});"'
```

Comment:
I'm not 100% sure, but I think the source string needs to include curly braces. Like `f'"{return rlnorm(...)}"'`

Reply:
it does not, I tested

Reply:
Cool. Sorry for the misdirection. The syntax here is different than for registering script blocks, which is why I was confused.